### PR TITLE
fixing the dev environment by downgrading clean-webpack-plugin

### PR DESCRIPTION
**Pull request by frehner**
creating an error boundary and also adding StrictMode

having all the components throw errors synchronously so they hit the error boundary

Honestly, I'm not sure I like the idea of this pattern or not. The goal behind it was to get something like the screenshot below for anyone who has an error, since asking people who have never devved/inspected plugins could be a tall task. But I'm not sold on it.

Thoughts?

Screen Shot 2019-06-21 at 3 49 59 PM

### CODE REVIEW

File: src/panel-app/ErrorBoundary.component.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            padding: "16px"
+          }}
+        >
+          <h1>single-spa Inspector broke!</h1>
+          <p>
+            Please close and repopen the devtools to get this working again.
+          </p>
+          <p>
+            Also, report this error{" "}
+            <a href="https://github.com/CanopyTax/single-spa-inspector/issues/new">
```
Comment (filoxo) [inline src/panel-app/ErrorBoundary.component.js]: It'd be pretty nice to automatically add some data to the URL which gets populated into the issue. See https://help.github.com/en/articles/about-automation-for-issues-and-pull-requests-with-query-parameters

File: package.json [hunk 2]
```diff
@@ -25,7 +26,7 @@
     "@babel/preset-env": "^7.4.5",
     "@babel/preset-react": "^7.0.0",
     "babel-loader": "^8.0.6",
-    "clean-webpack-plugin": "^3.0.0",
+    "clean-webpack-plugin": "^1.0.1",
```
Comment (joeldenning) [inline package.json]: why did clean-webpack-plugin@3 break things?

File: src/panel-app/useImportMapOverrides.js [hunk 3]
```diff
@@ -4,39 +4,69 @@ import { evalCmd } from "../inspected-window.helper.js";
...
-      return window.importMapOverrides.getOverrideMap()
-    })()`);
-    setOverrides(imports);
+    try {
+      const { imports } = await evalCmd(`(function() {
+        return window.importMapOverrides.getOverrideMap()
+      })()`);
+      setOverrides(imports);
+    } catch (err) {
+      err.message = `Error duing getImportMapOverrides. ${err.message}`;
```
Comment (joeldenning) [inline src/panel-app/useImportMapOverrides.js]: typo: `during`

File: src/panel-app/useImportMapOverrides.js [hunk 4]
```diff
@@ -4,39 +4,69 @@ import { evalCmd } from "../inspected-window.helper.js";
...
   async function removeOverride(currentMap) {
-    await evalCmd(`(function() {
-      return window.importMapOverrides.removeOverride("${currentMap}")
-    })()`);
+    try {
+      await evalCmd(`(function() {
+        return window.importMapOverrides.removeOverride("${currentMap}")
+      })()`);
+    } catch (err) {
+      err.message = `Error duing removeOverride. ${err.message}`;
```
Comment (joeldenning) [inline src/panel-app/useImportMapOverrides.js]: typo: `during`

File: src/panel-app/useImportMapOverrides.js [hunk 5]
```diff
@@ -4,39 +4,69 @@ import { evalCmd } from "../inspected-window.helper.js";
...
-    await Promise.all(overrideCalls);
-    await evalCmd(`window.location.reload()`);
+    try {
+      const overrideCalls = Object.entries(overrides).map(([map, url]) =>
+        !url ? removeOverride(map) : addOverride(map, url)
+      );
+      await Promise.all(overrideCalls);
+      await evalCmd(`window.location.reload()`);
+    } catch (err) {
+      err.message = `Error duing batchSetOverrides. ${err.message}`;
```
Comment (joeldenning) [inline src/panel-app/useImportMapOverrides.js]: typo: `during`

File: src/panel-app/useImportMapOverrides.js [hunk 6]
```diff
@@ -49,7 +79,12 @@ export default function useImportMapOverrides() {
       }
     }
 
-    initImportMapsOverrides();
+    try {
+      initImportMapsOverrides();
+    } catch (err) {
+      err.message = `Error duing initImportMapsOverrides. ${err.message}`;
```
Comment (joeldenning) [inline src/panel-app/useImportMapOverrides.js]: `during`

File: package.json (hunk 2 above)
Comment (frehner) [reply package.json]: Between the latest clean-webpack-plugin and copy-webpack-plugin, copy doesn’t actually copy assets on every compilation. Not sure who’s at fault here, but here’s a thread on it: https://github.com/webpack-contrib/copy-webpack-plugin/issues/385

File: src/panel-app/useImportMapOverrides.js (hunk 3 above)
Comment (frehner) [reply src/panel-app/useImportMapOverrides.js]: Haha dangit

**Comment by frehner** [comment]
addressed feedback.

**Comment by joeldenning** [comment]
👍

**Pull request merged**